internal/kubernetes/nodefilters.go

```diff
@@ -17,28 +17,48 @@ and limitations under the License.
 package kubernetes
 
 import (
+	"log"
 	"strings"
 	"time"
 
 	core "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/antonmedv/expr"
 )
 
-// NewNodeLabelFilter returns a filter that returns true if the supplied object
-// is a node with all of the supplied labels.
-func NewNodeLabelFilter(labels map[string]string) func(o interface{}) bool {
+// NewNodeLabelFilter returns a filter that returns true if the supplied node satisfies the boolean expression
+func NewNodeLabelFilter(expressionStr *string) (func(o interface{}) bool, error) {
+
 	return func(o interface{}) bool {
+		//This feels wrong but this is how the previous behavior worked so I'm only keeping it to maintain compatibility.
+		if *expressionStr == "" {
+			return true
+		}
+
 		n, ok := o.(*core.Node)
 		if !ok {
 			return false
 		}
-		for k, v := range labels {
-			if value, ok := n.GetLabels()[k]; value != v || !ok {
-				return false
-			}
+
+		nodeLabels := n.GetLabels()
+
+		parameters := make(map[string]interface{}, 8)
+
+		parameters["node_labels"] = nodeLabels
```

Comment:
I could see someone coming along with a use-case where anything in `metadata` is fair game. Heck, why not any part of the spec? :trollface:  So I don't think that needs to be implemented up front, but doing it in a forwards compatible manner would it be possible to have:

```
parameters["metadata"]["labels"] = nodeLabels
```

I threw this up in a go playground: https://play.golang.org/p/jrrQfyOH5P7

The goal would be that as use-cases come up for okaylisting parts of the node spec, there'd be an intuitive mapping between the shape of `kubectl get node <node> -o yaml` and our syntax.